store/sqlstore/post_store.go

```diff
@@ -600,54 +635,81 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, limit int
 	}
 
 	var posts, parents []*model.Post
-	_, err := s.GetReplica().Select(&posts,
-		`SELECT
-			*
-		FROM
-			Posts
-		WHERE
-			CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
-				AND ChannelId = :ChannelId
-				AND DeleteAt = 0
-		ORDER BY CreateAt `+sort+`
-		LIMIT :Limit
-		OFFSET :Offset`,
-		map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
-	if err != nil {
-		return nil, model.NewAppError("SqlPostStore.GetPostContext", "store.sql_post.get_posts_around.get.app_error", nil, "channelId="+channelId+err.Error(), http.StatusInternalServerError)
-	}
-
-	if len(posts) > 0 {
-		rootIds := []string{}
-		for _, post := range posts {
-			rootIds = append(rootIds, post.Id)
-			if post.RootId != "" {
-				rootIds = append(rootIds, post.RootId)
-			}
+	if skipRootFetch {
+		_, err := s.GetReplica().Select(&posts,
+			`SELECT
+			p.*,
+			(
+				SELECT
+					count(Posts.Id)
+				FROM
+					Posts
+				WHERE
+					(p.RootId != '' AND Posts.RootId = p.RootId) OR (p.RootId = '' AND Posts.RootId = p.Id)) AS ResponseCount
+			FROM
+				Posts p
+			WHERE
+					p.CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
+					AND p.ChannelId = :ChannelId
+					AND p.DeleteAt = 0
+			ORDER BY p.CreateAt `+sort+`
+			LIMIT :Limit
+			OFFSET :Offset`,
+			map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
+		if err != nil {
+			return nil, model.NewAppError("SqlPostStore.GetPostContext", "store.sql_post.get_posts_around.get.app_error", nil, "channelId="+channelId+err.Error(), http.StatusInternalServerError)
 		}
-
-		keys, params := MapStringsToQueryParams(rootIds, "PostId")
-
-		params["ChannelId"] = channelId
-		params["PostId"] = postId
-		params["Limit"] = limit
-		params["Offset"] = offset
-
-		_, err = s.GetReplica().Select(&parents,
+	} else {
```

Comment:
Duplicating the queries here seems error prone. Could we rewrite using squirrel and conditionally add the extra column fetch?